elements/lisk-transactions/test/15_pom_transaction.spec.ts

```diff
@@ -12,59 +12,494 @@
+					.networkIdentifier,
+		});
+
+		transactionWithScenario3 = new ProofOfMisbehaviorTransaction({
+			...validProofOfMisbehaviorTransactionScenario3.testCases.output,
+			networkIdentifier:
+				validProofOfMisbehaviorTransactionScenario1.testCases.input
+					.networkIdentifier,
+		});
+	});
+
 	describe('validateAsset', () => {
-		it.todo(
-			'when first height is greater than or equal to second height but equal maxHeighPrevoted it should not return errors',
-		);
+		it('should not return errors when first height is greater than or equal to second height but equal maxHeighPrevoted it ', () => {
+			const { errors, status } = transactionWithScenario1.validate();
+			expect(status).toBe(Status.OK);
+			expect(errors).toHaveLength(0);
+		});
 
-		it.todo(
-			"when height is greater than the second header's maxHeightPreviouslyForged it should not return errors",
-		);
+		it("should not return errors when height is greater than the second header's maxHeightPreviouslyForged", () => {
+			const { errors, status } = transactionWithScenario2.validate();
+			expect(status).toBe(Status.OK);
+			expect(errors).toHaveLength(0);
+		});
 
-		it.todo(
-			'when maxHeightPrevoted is greater than ther second maxHeightPrevoted it should not return errors',
-		);
+		it('should not return errors when maxHeightPrevoted is greater than ther second maxHeightPrevoted', () => {
+			const { errors, status } = transactionWithScenario3.validate();
+			expect(status).toBe(Status.OK);
+			expect(errors).toHaveLength(0);
+		});
 
-		it.todo('when headers are not contradicting it should return errors');
+		it('should return errors when headers are not contradicting', () => {
+			const nonContradictingTransaction = new ProofOfMisbehaviorTransaction({
+				...validProofOfMisbehaviorTransactionScenario1.testCases.output,
+				asset: {
+					header1:
+						validProofOfMisbehaviorTransactionScenario1.testCases.output.asset
+							.header1,
+					header2:
+						validProofOfMisbehaviorTransactionScenario1.testCases.output.asset
+							.header1,
+				},
+				networkIdentifier:
+					validProofOfMisbehaviorTransactionScenario1.testCases.input
+						.networkIdentifier,
+			});
 
-		it.todo('when headers are not properly signed it should return errors');
+			const { errors, status } = nonContradictingTransaction.validate();
+			expect(status).toBe(Status.FAIL);
+			expect(errors).toHaveLength(1);
+			expect(errors[0].message).toInclude(
+				'Blockheaders are identical. No contradiction detected.',
+			);
+		});
 	});
 
 	describe('applyAsset', () => {
-		it.todo('should add reward to balance of the sender');
-		it.todo('should deduct reward to balance of the misbehaving delegate');
+		let store: StateStoreMock;
+		let sender: Account;
+		let delegate: Account;
+
+		beforeEach(async () => {
+			sender = {
+				...defaultAccount,
+				...validProofOfMisbehaviorTransactionScenario1.testCases.input
+					.reportingAccount,
+				balance: BigInt(
```

Comment:
I think sender `nonce` is giving the error too.
it should use
`nonce: validProofOfMisbehaviorTransactionScenario1.testCases.output.nonce` for checking exact error you need